### PR TITLE
tests and N+1 queries 

### PR DESCRIPTION
**Pull request by Shakir-Hussain12**
**For this PR, I :**
- Used `includes` method to my posts and comments to solve N + 1 query issue.
- Added tests for Users' `Index` and `Show` page.
- Added tests for Posts' `Index` and `Show` page.
- Used Capybara and `:system` for all tests.
- Please use `bundle exec rspec` to run tests instead of the usual `rspec`.

### CODE REVIEW

File: spec/requests/posts/show_spec.rb [hunk 1]
```diff
@@ -1,20 +1,86 @@
...
-    it 'should render correct template' do
-      get '/users/123/posts/456'
-      expect(response.body).to render_template(:show)
+    it 'should get the post writer name' do
+      user_id = rand(1..2)
+      post_id = rand(1..5)
+      visit user_post_path(user_id, post_id)
+      container = page.find('div', class: 'top-com')
+      title = container.find('h1').text.split.last
+      expect(title).to eq(User.find(user_id).name)
```
Comment (AnselemOdims) [inline spec/requests/posts/show_spec.rb]: - [x] Good job adding all these tests, but please kindly include test that asserts you can see the post's title.

**Review by AnselemOdims — CHANGES_REQUESTED** [review]
### Hi @team, 
You have been doing a tremendous job so far :ok_hand:
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 🎉
PR has a title and summary. :heavy_check_mark:
No linter errors :heavy_check_mark:

## Required Changes ♻️
- [ ] Please check inline comments under the review

## Optional suggestions
_Comments with the [OPTIONAL] prefix are not a must to implement however, It's strongly recommended to work on them as they can make your code better._

## Happy coding!👏👏👏

#### Please leave any questions or comments in the PR if you have any doubts or unclarity! You can tag me using @AnselemOdims for a quicker response

**_Note: Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by danmainah — APPROVED** [review]
Hi @Team, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 🎉 🎉 

## Highlights
✔️ Linter setup correctly and all checks are passing
✔️ All tests added successfully

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by danmainah — APPROVED** [review]
Hi @Team, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 🎉 🎉 

## Highlights
✔️ Linter setup correctly and all checks are passing
✔️ All tests added successfully

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**